common/src/main/java/com/revenuecat/purchases/common/ReceiptInfo.kt

```diff
@@ -53,3 +69,38 @@ class ReceiptInfo(
             "duration=$duration)"
     }
 }
+
+internal fun StoreProduct.platformProductId(): PlatformProductId {
+    return PlatformProductId(id)
+}
+
+internal fun SubscriptionOption.platformProductId(): PlatformProductId? {
+    return when (this) {
+        is GoogleSubscriptionOption -> GooglePlatformProductId(
+            productId,
+            basePlanId,
+            offerId
+        )
+        else -> null
+    }
+}
```

Comment:
I don't love this type checking here for `GoogleSubscriptionModel` but I wanted this interface to be internal on `StoreProduct` and `SubscriptionOption`.

`StoreProduct` will always just use `PlatformProductId` for Amazon (OTP and sub) and for Google OPT.

`GoogleSubscriptionOption` is the one that needs the product id, base plan id, and offer id with the `GooglePlatformProductId` object. This `ReceiptInfo` and the models are in separate modules so there was no way to make this **internal on the models**. 

So this feels the cleanest way to not expose any of this data 🤷‍♂️

Reply:
Liskov won't be super happy but it's not the end of the world. 
Can this be handled at the subclass level, though?

Reply:
I tried the subclass as my first approach (you can see in one of the commits) but it required stuff to be made public because of the way our modules are setup 😞

Reply:
👍  cc @tonidero @vegaro even more reasons to move away from the current module setup, we should take care of that sooner rather than later

Reply:
I'm not sure why you wouldn't make these public... they are part of the platform specific information and can be made available for devs. They're not necessarily needed, but I don't see any issue in people downcasting StoreProducts/SubscriptionOptions and getting the basePlan, offerId etc...
I think an approach like this would be cleaner: https://github.com/RevenueCat/purchases-android/pull/920/files